utils/webpack/webpack.config.js

```diff
@@ -0,0 +1,207 @@
+const path = require('path');
+const webpack = require('webpack');
+const chalk = require('chalk');
+const TerserPlugin = require('terser-webpack-plugin');
+const HTMLWebpackPlugin = require('html-webpack-plugin');
+const ProgressBarWebpackPlugin = require('progress-bar-webpack-plugin');
+const ShopgateIndexerPlugin = require('./plugins/ShopgateIndexerPlugin');
+const { ENV, isDev, PUBLIC_FOLDER } = require('./lib/variables');
+const getAppSettings = require('./lib/getAppSettings');
+const convertLanguageToISO = require('./lib/convertLanguageToISO');
+const getComponentsSettings = require('./lib/getComponentsSettings');
+const getThemeConfig = require('./lib/getThemeConfig');
+const getThemeLanguage = require('./lib/getThemeLanguage');
+const getDevConfig = require('./lib/getDevConfig');
+const i18n = require('./lib/i18n');
+const getExtensionsNodeModulesPaths = require('./lib/getExtensionsNodeModulesPaths');
+
+const themePath = process.cwd();
+const appConfig = getAppSettings(themePath);
+const componentsConfig = getComponentsSettings(themePath);
+const themeConfig = getThemeConfig(themePath, appConfig);
+const isoLang = convertLanguageToISO(appConfig.language);
+const { sourceMap, ip, apiPort } = getDevConfig();
+const t = i18n(__filename);
+
+const config = {
+  mode: ENV,
+  entry: {
+    app: [
+      path.resolve(__dirname, 'lib', 'polyfill.js'),
+      `${themePath}/index.jsx`,
+    ],
+    common: [
+      'intl',
+      `intl/locale-data/jsonp/${isoLang}`,
+      'react',
+      'react-dom',
+      'glamor',
+      'react-redux',
+      'reselect',
+    ],
+  },
+  output: {
+    filename: !isDev ? '[name].[hash].js' : '[name].js',
+    chunkFilename: '[name].[chunkhash].js',
+    path: path.resolve(themePath, PUBLIC_FOLDER),
+    publicPath: isDev ? '/' : (process.env.publicPath || './'),
+  },
+  resolve: {
+    extensions: ['.json', '.js', '.jsx'],
+    alias: {
+      'react-dom': '@hot-loader/react-dom',
+    },
+    modules: [
+      path.resolve(themePath, 'widgets'),
+      path.resolve(themePath, 'node_modules'),
+      path.resolve(themePath, '..', '..', 'node_modules'),
+      path.resolve(themePath, '..', '..', 'extensions'),
+      ...getExtensionsNodeModulesPaths(),
+    ],
+  },
+  plugins: [
```

Comment:
I see that some modules where removed:
- ScriptExtHtmlWebpackPlugin
- StringReplacePlugin
- CompressionWebpackPlugin (only dev)
- GenerateSW (only dev)

what are the reasons?

Reply:
No reason. Just missed them. I added them now.